x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/analysis/analyzer/VerifierErrorMessagesTests.java

```diff
@@ -1135,4 +1135,13 @@ public void testErrorMessageForMatrixStatsWithScalars() {
         assertEquals("1:17: [SKEWNESS()] cannot be used on top of operators or scalars",
                 error("SELECT SKEWNESS(ABS(int * 10.123)) FROM test"));
     }
+
+    public void testCastOnInexact() {
+        // inexact with underlying keyword
+        assertEquals("1:36: [some.string] of data type [text] cannot be used for [CAST()] inside WHERE clause",
+                error("SELECT * FROM test WHERE NOT (CAST(some.string AS string) = 'foo') OR true"));
```

Comment:
Does the relatively not-straight `WHERE` clause bare any meaning to the test target (for instance, vs. some `WHERE CAST(some.string AS string) = 'foo'`)? Just curious.

Reply:
Yes, produces more complex tree, so the error can be caught even in such cases: (`forEachUp/Down`, `forEachExpressionUp/Down`, etc).